cmake/modules/FindFFMPEG.cmake

```diff
@@ -209,19 +209,17 @@ endif()
 # Internal FFMPEG
 if(NOT FFMPEG_FOUND)
   include(ExternalProject)
-  file(STRINGS ${CMAKE_SOURCE_DIR}/tools/depends/target/ffmpeg/FFMPEG-VERSION VER)
-  string(REGEX MATCH "VERSION=[^ ]*$.*" FFMPEG_VER "${VER}")
-  list(GET FFMPEG_VER 0 FFMPEG_VER)
-  string(SUBSTRING "${FFMPEG_VER}" 8 -1 FFMPEG_VER)
-  string(REGEX MATCH "BASE_URL=([^ ]*)" FFMPEG_BASE_URL "${VER}")
-  list(GET FFMPEG_BASE_URL 0 FFMPEG_BASE_URL)
-  string(SUBSTRING "${FFMPEG_BASE_URL}" 9 -1 FFMPEG_BASE_URL)
+  include(cmake/scripts/common/ModuleHelpers.cmake)
+
+  get_archive_name(ffmpeg)
 
   # allow user to override the download URL with a local tarball
   # needed for offline build envs
   if(FFMPEG_URL)
     get_filename_component(FFMPEG_URL "${FFMPEG_URL}" ABSOLUTE)
   else()
+    # github tarball format is tagname.tar.gz (eg 4.4-N-Alpha1.tar.gz)
+    # tagname is our FFMPEG_VER from VERSION file.
     set(FFMPEG_URL ${FFMPEG_BASE_URL}/archive/${FFMPEG_VER}.tar.gz)
```

Comment:
This frustrates me a little. as we are using github storage for the tarball, the tarball is generated with just the tagname.tar.gz.

Im trying to avoid doing to many edge cases with what we provide from the function. The way i see it, we can go with this, that has a hardcoded extension, or alternatively we can strip out ffmpeg from ${ARCHIVE} into a variable and use that.
As it stands, github currently only provides tar.gz or zip extensions to downloads, so is it safe enough to just assume, and therefore hardcode, tar.gz?

Not sure whats best?

Reply:
The following would be the alternative. It feels just as messy, hardcoding ffmpeg-, as arithmetic on variables is even more messy in cmake (eg MATH(EXPR len "${len}+1"))

```
    # github tarball format is tagname.tar.gz (eg 4.4-N-Alpha1.tar.gz)
    # tagname is our FFMPEG_VER from VERSION file. extract from ${ARCHIVE}
    string(LENGTH "ffmpeg-" len)
    string(SUBSTRING ${ARCHIVE} ${len} -1 VER_ARCHIVE)
    set(FFMPEG_URL ${FFMPEG_BASE_URL}/archive/${VER_ARCHIVE})
```

Reply:
What if we download the tarball, name it properly, upload it to our mirrors then fetch it from there?

Reply:
Would make my life easier, not sure if there would be any fallout in such a change

Reply:
I'd keep it as is, it's unlikely that github stops providing tar.gz anytime soon.

Reply:
Ok, for simplicity sake, we'll just go with the hardcoded tar.gz for now. Can always revisit this later if we change the ffmpeg distribution method